src/Waives.Pipelines/WorkPool.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Waives.Pipelines
+{
+    internal class WorkPool : IDisposable
+    {
+        private readonly int _maxConcurrency;
```

Comment:
💪 The `MaxConcurrency` constant should probably be defined here rather than... wherever it is now defined.